code/WorkInProgress/MechanicMadness.dm

```diff
@@ -15,6 +15,235 @@
+// mechanics containers for mechanics components (read: portable horn [read: vuvuzela] honkers! yaaaay!)
+//
+/obj/item/storage/mechanics // generic
+	name="Generic MechComp Housing"
+	desc="You should not bee seeing this! Call 1-800-CODER or just crusher it"
+	icon='icons/misc/mechanicsExpansion.dmi'
+	can_hold=list(/obj/item/mechanics)
+	deconstruct_flags = 0 //nope, so much nope.
+	slots=1
+	var/open=1
+	var/welded=0
+	var/can_be_welded=0
+	var/can_be_anchored=0
+	custom_suicide=1
+	ex_act(severity)
+		switch(severity)
+			if (1.0)
+				src.dispose() // disposing upon being blown up unlike all those decorative rocks on cog2
+				return
+			if (2.0)
+				//tbd, maybe unbolt/open?
+				return
+			if (3.0)
+				//??????
+				return
+		return
+	suicide(var/mob/user as mob) // lel
+		if (!src.user_can_suicide(user))
+			return 0
+		user.visible_message("<span class='alert'><b>[user] stares into the [src], trying to make sense of its function!</b></span>")
+		SPAWN_DBG(3 SECONDS)
+			user.visible_message("<span class='alert'><b>[user]'s brain melts!</b></span>")
+			playsound(get_turf(user), "sound/weapons/phaseroverload.ogg", 100)
+			user.take_brain_damage(69*420)
+		SPAWN_DBG(20 SECONDS)
+			if (user && !isdead(user))
+				user.suiciding = 0
+		return
+
+
+	attackby(obj/item/W as obj, mob/user as mob)
+
+		if (isscrewingtool(W))
+			if(src.welded)
+				boutput(user,"<span class='alert'>The [src] is welded shut.</span>")
+				return
+			src.does_not_open_in_pocket=src.open
+			src.open=!src.open
+			playsound(src.loc,'sound/items/screwdriver.ogg',50)
+			boutput(user,"<span class='notice'>You [src.open ? "unsecure" : "secure"] the [src]'s cover</span>")
+			src.updateIcon()
+			return
+		else if (iswrenchingtool(W))
+			if(!src.can_be_anchored)
+				boutput(user,"<span class='alert'>[src] cannot be anchored to the ground.</span>")
+				return
+			if(!src.open || src.welded)
+				boutput(user,"<span class='alert'>You are unable to access the [src]'s bolts as they are on the inside.</span>")
+				return
+			if(!isturf(src.loc) && !src.anchored)
+				boutput(user,"<span class='alert'>You cannot anchor a component housing inside something else.</span>")
+				return
+			src.anchored=!src.anchored
+			playsound(src.loc,'sound/items/Ratchet.ogg',50)
+			boutput(user,"<span class='notice'>You [src.anchored ? "anchor the [src] to" : "unsecure the [src] from"] the ground</span>")
+			if (!src.anchored)
+				src.destroy_outside_connections() //burn those bridges
+			return
+		else if (isweldingtool(W))
+			if (!src.can_be_welded)
+				boutput(user,"<span class='alert'>[src]'s cover cannot be welded shut.</span>")
+				return
+			if (src.open)
+				boutput(user,"Why would you want to weld something <i>open?</i>")
+				return
+			if(W:try_weld(user, 1))
+				src.welded=!src.welded
+				boutput(user,"<span class='notice'>You [src.welded ? "" : "un"]weld the [src]'s cover</span>")
+				src.updateIcon()
+				return
+		else
+			..()
+		return
```

Comment:
Generally, if attackby() has been "successfully" handled, you want to `return 1`